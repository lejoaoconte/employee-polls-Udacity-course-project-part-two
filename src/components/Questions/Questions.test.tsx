import "@testing-library/jest-dom";

import { fireEvent, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import { Questions } from ".";

const users = {
  sarahedo: {
    id: "sarahedo",
    password: "password123",
    name: "Sarah Edo",
    avatarURL: "https://github.com/sarah.png",
    answers: {
      "8xf0y6ziyjabvozdd253nd": "optionOne",
      "6ni6ok3ym7mf1p33lnez": "optionOne",
      am8ehyc8byjqgar0jgpub9: "optionTwo",
      loxhs1bqm25b708cmbf3g: "optionTwo",
    },
    questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
  },
};

const questions = {
  "8xf0y6ziyjabvozdd253nd": {
    id: "8xf0y6ziyjabvozdd253nd",
    author: "sarahedo",
    timestamp: 1467166872634,
    optionOne: {
      votes: ["sarahedo"],
      text: "Build our new application with Javascript",
    },
    optionTwo: {
      votes: [],
      text: "Build our new application with Typescript",
    },
  },
};

describe("Questions", () => {
  const setup = () => {
    const component = render(
      <MemoryRouter>
        <Questions
          questions={Object.values(questions)}
          users={Object.values(users)}
        />
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();

    const questionsComponent = component.getByTestId("questions-test");
    const questionComponentItem = component.getByTestId("questions-item-test");
    const questionItemRender = component.getByText(/Sarah/);

    return {
      questionsComponent,
      questionComponentItem,
      questionItemRender,
      ...component,
    };
  };

  it("Testing render on screen", () => {
    const { questionsComponent, questionComponentItem } = setup();
    expect(questionsComponent).toBeInTheDocument;
    expect(questionComponentItem).toBeInTheDocument;
  });

  it("Test render infos on screen", () => {
    const { questionItemRender } = setup();
    expect(questionItemRender).toBeInTheDocument;
  });

  it("Test onClick element", () => {
    const { questionComponentItem } = setup();

    fireEvent.click(questionComponentItem);
  });
});
