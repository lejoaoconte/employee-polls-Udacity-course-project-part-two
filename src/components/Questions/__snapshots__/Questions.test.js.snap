// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions Test onClick element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-bcXHqe jxQawA"
        data-testid="questions-test"
      >
        <ul
          class="sc-gswNZR cpBOMx"
        >
          <li
            data-testid="questions-item-test"
          >
            <img
              alt="user"
            />
            <p>
              Created by
               
              <span>
                Sarah Edo
              </span>
            </p>
            <span
              class="time"
            >
              06/28/2016 11:21 PM
            </span>
          </li>
        </ul>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-bcXHqe jxQawA"
      data-testid="questions-test"
    >
      <ul
        class="sc-gswNZR cpBOMx"
      >
        <li
          data-testid="questions-item-test"
        >
          <img
            alt="user"
          />
          <p>
            Created by
             
            <span>
              Sarah Edo
            </span>
          </p>
          <span
            class="time"
          >
            06/28/2016 11:21 PM
          </span>
        </li>
      </ul>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Questions Test render infos on screen 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-bcXHqe jxQawA"
        data-testid="questions-test"
      >
        <ul
          class="sc-gswNZR cpBOMx"
        >
          <li
            data-testid="questions-item-test"
          >
            <img
              alt="user"
            />
            <p>
              Created by
               
              <span>
                Sarah Edo
              </span>
            </p>
            <span
              class="time"
            >
              06/28/2016 11:21 PM
            </span>
          </li>
        </ul>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-bcXHqe jxQawA"
      data-testid="questions-test"
    >
      <ul
        class="sc-gswNZR cpBOMx"
      >
        <li
          data-testid="questions-item-test"
        >
          <img
            alt="user"
          />
          <p>
            Created by
             
            <span>
              Sarah Edo
            </span>
          </p>
          <span
            class="time"
          >
            06/28/2016 11:21 PM
          </span>
        </li>
      </ul>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Questions Testing render on screen 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-bcXHqe jxQawA"
        data-testid="questions-test"
      >
        <ul
          class="sc-gswNZR cpBOMx"
        >
          <li
            data-testid="questions-item-test"
          >
            <img
              alt="user"
            />
            <p>
              Created by
               
              <span>
                Sarah Edo
              </span>
            </p>
            <span
              class="time"
            >
              06/28/2016 11:21 PM
            </span>
          </li>
        </ul>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-bcXHqe jxQawA"
      data-testid="questions-test"
    >
      <ul
        class="sc-gswNZR cpBOMx"
      >
        <li
          data-testid="questions-item-test"
        >
          <img
            alt="user"
          />
          <p>
            Created by
             
            <span>
              Sarah Edo
            </span>
          </p>
          <span
            class="time"
          >
            06/28/2016 11:21 PM
          </span>
        </li>
      </ul>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
